File: backend/app/models/validator.py
from typing import Literal

from pydantic import BaseModel


class ValidateCodeRequest(BaseModel):
    codes: list[str]


class ValidateResult(BaseModel):
    code: str
    status: Literal["valid", "not-found", "deprecated"]
    term: str | None = None
    ontology: str | None = None


class ValidateResponse(BaseModel):
    results: list[ValidateResult]

File: frontend/src/types/validator.ts
export type ValidateStatus = 'valid' | 'not-found' | 'deprecated';

export interface ValidateCodeRequest {
  codes: string[];
}

export interface ValidateCodeResult {
  code: string;
  status: ValidateStatus;
  term?: string;
  ontology?: string;
}

export interface ValidateCodeResponse {
  results: ValidateCodeResult[];
}
